d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -211,8 +219,102 @@ public void onSuccess(None result)
         }
         runNextMarkUpOrMarkDown();
       }
-    });
+    };
+
+    final Callback<None> doMarkUpCallback = new Callback<None>() {
+      @Override
+      public void onError(Throwable e)
+      {
+        // this would ideally never happen
+        callback.onError(e);
+      }
+
+      @Override
+      public void onSuccess(None result)
+      {
+        _log.info("doing mark up of main cluster " + _cluster);
```

Comment:
"main" cluster is potentially confusing here.  How about "primary" or "live" or "regular" something similar?

Reply:
thanks for pointing this. I have deleted the log statement and only log if mark up succeeds or fails and using the name of the cluster instead of "main", to avoid confusion.